tests/net/lib/http_server/tls/src/main.c

```diff
@@ -0,0 +1,259 @@
+ *
+ * This is the public key of the server.
+ */
+static const unsigned char server[] = {
+#include "server.inc"
+};
+
+/**
+ * @brief The Server Private Key
+ *
+ * This is the private key of the server.
+ */
+static const unsigned char server_privkey[] = {
+#include "server_privkey.inc"
+};
+
+/**
+ * @brief The Client Certificate
+ *
+ * This is the public key of the client.
+ */
+static const unsigned char client[] = {
+#include "client.inc"
+};
+
+/**
+ * @brief The Client Private Key
+ *
+ * This is the private key of the client.
+ */
+static const unsigned char client_privkey[] = {
+#include "client_privkey.inc"
+};
+#else /* CONFIG_TLS_CREDENTIALS */
+#define ca             NULL
+#define server         NULL
+#define server_privkey NULL
+#define client         NULL
+#define client_privkey NULL
+#endif /* CONFIG_TLS_CREDENTIALS */
+
+static void server_thread_fn(void *arg0, void *arg1, void *arg2)
+{
+	struct http_server_ctx *ctx = (struct http_server_ctx *)arg0;
+
+	int r;
+	int client_fd;
+	socklen_t addrlen;
+	char addrstr[BUFFER_SIZE];
+	struct sockaddr_in sa;
+
+	k_thread_name_set(k_current_get(), "server");
+
+	memset(&sa, 0, sizeof(sa));
+	addrlen = sizeof(sa);
+
+	k_sem_give(&server_sem);
+
+	client_fd = accept_new_client(ctx->server_fd);
+
+	NET_DBG("calling recv()");
+	r = recv(client_fd, addrstr, sizeof(addrstr), 0);
+	zassert_not_equal(r, -1, "recv() failed (%d)", errno);
+	zassert_equal(r, SECRET_SIZE, "expected: %zu actual: %d", SECRET_SIZE, r);
+
+	NET_DBG("calling send()");
+	r = send(client_fd, SECRET, SECRET_SIZE, 0);
+	zassert_not_equal(r, -1, "send() failed (%d)", errno);
+	zassert_equal(r, SECRET_SIZE, "expected: %zu actual: %d", SECRET_SIZE, r);
+
+	r = close(client_fd);
+	zassert_not_equal(r, -1, "close() failed on the server fd (%d)", errno);
+}
+
+static void test_tls(void)
+{
+	int r;
+	int client_fd;
+	int proto = IPPROTO_TCP;
+	char *addrstrp;
+	k_tid_t server_thread_id;
+	struct sockaddr_in sa;
+	char addrstr[256];
+
+	k_sem_init(&server_sem, 0, 1);
+
+	/* set the common protocol for both client and server */
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+		proto = IPPROTO_TLS_1_2;
+	}
+
+	struct http_server_ctx ctx;
+
+	int server_fd = http_server_init(&ctx);
+
+	zassert_true(server_fd >= 0, "Failed to create server socket");
+
+	server_thread_id =
+		k_thread_create(&server_thread, server_stack, STACK_SIZE, server_thread_fn, &ctx,
+				NULL, NULL, K_PRIO_PREEMPT(8), 0, K_NO_WAIT);
+
+	r = k_sem_take(&server_sem, K_MSEC(TIMEOUT));
+	zassert_equal(0, r, "failed to synchronize with server thread (%d)", r);
+
+	k_thread_name_set(k_current_get(), "client");
+
+	r = socket(AF_INET, SOCK_STREAM, proto);
+	zassert_not_equal(r, -1, "failed to create client socket (%d)", errno);
+	client_fd = r;
+
+	if (IS_ENABLED(CONFIG_TLS_CREDENTIALS) && IS_ENABLED(CONFIG_NET_SOCKETS_SOCKOPT_TLS)) {
+
+		static const sec_tag_t client_tag_list_verify_none[] = {
+			HTTP_SERVER_CA_CERTIFICATE_TAG,
```

Comment:
Can't build this test as these macros are missing. Is there some config missing to enable?

Reply:
@nxpadamm sorry for late reply, I was busy with other projects. I will try to fix this soonish.